sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -485,8 +487,16 @@ trait InputRDDCodegen extends CodegenSupport {
       ""
     }
     s"""
-       | while ($limitNotReachedCond $input.hasNext()) {
+       | long $numOutputRows = 0;
+       | boolean $taskInterrupted = false;
+       | while ($limitNotReachedCond !$taskInterrupted && $input.hasNext()) {
        |   InternalRow $row = (InternalRow) $input.next();
+       |   $numOutputRows++;
+       |   if ($numOutputRows % 1000 == 0) {
```

Comment:
`1000` looks like a magic number. So, we don't need to interrupt when the number of rows is less than 1000 because it will finish quickly in any way?

Reply:
Yea, it's by intuition. And yes we expect 1000 rows to finish quickly soon.